sky/backends/backend_utils.py

```diff
@@ -644,11 +645,16 @@ def write_cluster_config(
         keep_launch_fields_in_existing_config: bool = True) -> Dict[str, str]:
     """Fills in cluster configuration templates and writes them out.
 
-    Returns: {provisioner: path to yaml, the provisioning spec}.
-      'provisioner' can be
-        - 'ray'
-        - 'tpu-create-script' (if TPU is requested)
-        - 'tpu-delete-script' (if TPU is requested)
+    Returns:
+        Dict with the following keys:
+        - 'ray': Path to the generated Ray yaml config file
+        - 'cluster_name': Name of the cluster
+        - 'cluster_name_on_cloud': Name of the cluster as it appears in the
+          cloud provider
+        - 'config_hash': Hash of the cluster config and file mounts
+          contents. Can be missing if we failed to calculate the hash for some
+          reason
```

Comment:
```suggestion
        - 'config_hash': Hash of the cluster config and file mounts
          contents. Can be missing if we fail to calculate the hash for some
          reason, which is fine, i.e., we do not do the provisioning
          optimization.
```

Reply:
I think we should not expect this to fail. It's fine with me to catch the exception in case it does fail, to avoid blocking the user, but it is still bad and unexpected, and will lead to degraded performance for the user.

Reply:
Yep, it should not fail. I guess we can be conservative here, considering this is not in the critical path of the provisioning, exceptions can be tolerable?